compiler/rustc_codegen_llvm/src/intrinsic.rs

```diff
@@ -334,8 +334,11 @@ impl IntrinsicCallMethods<'tcx> for Builder<'a, 'll, 'tcx> {
         self.call(expect, &[cond, self.const_bool(expected)], None)
     }
 
-    fn sideeffect(&mut self, unconditional: bool) {
-        if unconditional || self.tcx.sess.opts.debugging_opts.insert_sideeffect {
+    fn sideeffect(&mut self) {
+        // This kind of check would make a ton of sense in the caller, but currently the only
+        // caller of this function is in `rustc_codegen_ssa`, which is agnostic to whether LLVM
+        // codegen backend being used, and so is unable to check the LLVM version.
+        if llvm_util::get_version() < (12, 0, 0) {
```

Comment:
This should be called pretty rarely I think, but I am a bit worried about the performance - get_version makes 3 separate FFI calls after all. We'll see what perf says though.

Reply:
Yeah, we can just call the LLVMRustGetMajorVersion directly here.